dashboards/key_metrics/views/accounts.py

```diff
@@ -0,0 +1,199 @@
+from datetime import datetime, timedelta
+
+import streamlit as st
+import pandas as pd
+
+from dashboards.utils.charts import chart_bars, chart_lines
+from dashboards.utils.date_utils import get_start_date
+from dashboards.key_metrics.constants import (
+    SUPPORTED_CHAINS_CORE,
+    SUPPORTED_CHAINS_PERPS,
+)
+
+st.markdown("# Accounts Activity")
+
+if "chain" not in st.session_state:
+    st.session_state.chain = st.query_params.get("chain", "all")
+if "date_range" not in st.session_state:
+    st.session_state.date_range = st.query_params.get("date_range", "30d")
+
+st.query_params.chain = st.session_state.chain
+st.query_params.date_range = st.session_state.date_range
+
+
+@st.cache_data(ttl="30m")
+def fetch_data(date_range, chain):
+    end_date = datetime.now()
+    start_date = get_start_date(date_range)
+
+    chains_to_fetch = [*SUPPORTED_CHAINS_CORE] if chain == "all" else [chain]
+
+    core_account_activity_daily = [
+        st.session_state.api.get_core_account_activity(
+            start_date=start_date.date(),
+            end_date=end_date.date(),
+            chain=current_chain,
+            resolution="day",
+        )
+        for current_chain in chains_to_fetch
+        if current_chain in SUPPORTED_CHAINS_CORE
+    ]
+    core_account_activity_monthly = [
+        st.session_state.api.get_core_account_activity(
+            start_date=start_date.date(),
+            end_date=end_date.date(),
+            chain=current_chain,
+            resolution="month",
+        )
+        for current_chain in chains_to_fetch
+        if current_chain in SUPPORTED_CHAINS_CORE
+    ]
+    core_nof_stakers = [
+        st.session_state.api.get_core_nof_stakers(
+            start_date=start_date.date(),
+            end_date=end_date.date(),
+            chain=current_chain,
+        )
+        for current_chain in chains_to_fetch
+        if current_chain in SUPPORTED_CHAINS_CORE
+    ]
+    perps_account_activity_daily = [
+        st.session_state.api.get_perps_account_activity(
+            start_date=start_date.date(),
+            end_date=end_date.date(),
+            chain=current_chain,
+            resolution="day",
+        )
+        for current_chain in chains_to_fetch
+        if current_chain in SUPPORTED_CHAINS_PERPS
+    ]
+    perps_account_activity_monthly = [
+        st.session_state.api.get_perps_account_activity(
+            start_date=start_date.date(),
+            end_date=end_date.date(),
+            chain=current_chain,
+            resolution="month",
+        )
+        for current_chain in chains_to_fetch
+        if current_chain in SUPPORTED_CHAINS_PERPS
+    ]
+
+    return {
+        "core_account_activity_daily": (
+            pd.concat(core_account_activity_daily, ignore_index=True)
```

Comment:
might be cleaner to just wrap the calls above with `pd.concat`, and avoid the if statement here. not actually sure if it works to create a default `pd.DataFrame()`

Reply:
it doesn't work